sql/plan/alter_foreign_key.go

```diff
@@ -0,0 +1,168 @@
+package plan
+
+import (
+	"fmt"
+	"github.com/liquidata-inc/go-mysql-server/sql"
+	"gopkg.in/src-d/go-errors.v1"
+	"strings"
+)
+
+var (
+	// ErrNoForeignKeyAlterSupport is returned when the table does not support ALTER TABLE ... FOREIGN KEY operations.
+	ErrNoForeignKeyAlterSupport = errors.NewKind("the table does not support foreign key alter operations")
+	// ErrForeignKeyMissingColumns is returned when an ALTER TABLE ADD FOREIGN KEY statement does not provide any columns
+	ErrForeignKeyMissingColumns = errors.NewKind("cannot create a foreign key without columns")
```

Comment:
This would ideally be handled by the parser as a syntax error. See if that's easily doable.

Reply:
Already errors on it in the parser, but I'll retain the error here.